fix(browser_adapter.ts): baseElement.getAttribute

PatrickJS opened the pull request:
currently throwing an error

``` error
Error during instantiation of LocationStrategy! (RouterLink -> Router -> Location -> LocationStrategy). 
ORIGINAL ERROR: TypeError: baseElement.attr is not a function
```

closes #3441

tbosch commented on modules/angular2/src/dom/browser_adapter.ts:
```diff
@@ -287,7 +287,7 @@ function getBaseElementHref(): string {
     }
   }
-  return baseElement.attr('href');
+  return baseElement.getAttribute('href');
```
Hey, could you add a unit test for this?

gdi2290 replied:
I added tests but ran into a problem with [baseElement cache](https://github.com/angular/angular/pull/3214/files#r35475994)

tbosch replied:
Could you add a `resetBaseElement` method to the `DomAdapter` (and all implementations). You can narrow it down to only the JS/Dart adapter via `if (DOM.supportsDOMEvents()) {...}`

gdi2290 replied:
I think I did something wrong here when adding `resetBaseElement` and `getBaseHref` to the other adapters

gdi2290 commented on modules/angular2/test/dom/dom_adapter_spec.ts:
```diff
@@ -49,5 +49,49 @@ export function main() {
+      var $head = DOM.defaultDoc().head;
+      DOM.appendChild($head, $base);
+
+      var subject = DOM.getBaseHref();
```
this fails due to [`baseElement being cache`](https://github.com/gdi2290/angular/blob/patch-1/modules/angular2/src/dom/browser_adapter.ts#L285) from the prevous test

gdi2290 commented:
I'm having trouble correctly testing this in Dart. Can anyone help?

btford commented:
@gdi2290 sure– what's the problem?

gdi2290 commented:
@btford with Dart I'm trying not run these tests for Dart (since it's only in js atm).  Using `if (DOM.supportsDOMEvents)` [travis dart error](https://github.com/angular/angular/pull/3214/files#diff-b4743922539bf28ebff17607917d2620R53) doesn't seem to work when avoiding the Dart version. I recall an environment variable (`IS_DARTIUM`) that is used to narrow down a test to js so should I use that rather than `supportsDOMEvents` as Tobias suggested? Unless he meant for me to write out `getBaseHref` for Dart too (I know he is on vaca this week)

btford commented:
> Unless he meant for me to write out getBaseHref for Dart too (I know he is on vaca this week)

I think that's what he meant, but I'm not sure. :/

btford commented:
Moving this to alpha-35.

PatrickJS committed:
fix(browser_adapter.ts): baseElement.getAttribute

currently throwing an error
```error
Error during instantiation of LocationStrategy! (RouterLink -> Router
-> Location -> LocationStrategy).
ORIGINAL ERROR: TypeError: baseElement.attr is not a function
```

gdi2290 commented:
@btford can you take over this PR? you can either close and recreate it or push to my repo `git push patrickjs yourbranch:patch-1` for it to show up in this PR. All that's left is Dart support and I don't want this PR to live through another version :/

btford commented:
Yes, I'll get this in today.

btford commented:
Fixed and pushed to presubmit. :+1:

gdi2290 commented:
thanks! :+1:
